js/block-editor.js

```diff
@@ -50,56 +56,77 @@ function getCurrentLanguage() {
 				// Need to wait the ajax response before triggering the block editor post save action.
-				var data = {
-					action:     'post_lang_choice',
-					lang:       event.target.value,
-					post_type:  $( '#post_type' ).val(),
-					post_id:    $( '#post_ID' ).val(),
-					_pll_nonce: $( '#_pll_nonce' ).val()
+				if ( $( this ).data( 'old-value' ) !== selectedOption.value && ! emptyPost ) {
+					dialog.dialog( 'open' );
+				} else {
+					// Block editor doesn't allow to save an empty post.
+					// So we must revert the language to the old one because the browser asks if we want to quit the page if we trigger its reloading.
+					selectedOption.value = $( this ).data( 'old-value' );
 				}
 
-				$.post(
-					ajaxurl,
-					data,
-					function( response ) {
-						var res = wpAjax.parseAjaxResponse( response, 'ajax-response' );
-						$.each(
-							res.responses,
-							function() {
-								switch ( this.what ) {
-									case 'translations': // Translations fields
-										// Data is built and come from server side and is well escaped when necessary
-										$( '.translations' ).html( this.data ); // phpcs:ignore WordPressVIPMinimum.JS.HTMLExecutingFunctions.html
-										init_translations();
-									break;
-									case 'flag': // Flag in front of the select dropdown
-										// Data is built and come from server side and is well escaped when necessary
-										$( '.pll-select-flag' ).html( this.data ); // phpcs:ignore WordPressVIPMinimum.JS.HTMLExecutingFunctions.html
-									break;
-								}
+				dialogResult.then(
+					() => {
+						var data = {
+							action:     'post_lang_choice',
+							lang:       selectedOption.value,
+							post_type:  $( '#post_type' ).val(),
+							post_id:    $( '#post_ID' ).val(),
+							_pll_nonce: $( '#_pll_nonce' ).val()
+						}
+
+						$.post(
+							ajaxurl,
+							data,
+							function( response ) {
+								var res = wpAjax.parseAjaxResponse( response, 'ajax-response' );
+								$.each(
+									res.responses,
+									function() {
+										switch ( this.what ) {
+											case 'translations': // Translations fields
+												// Data is built and come from server side and is well escaped when necessary
+												$( '.translations' ).html( this.data ); // phpcs:ignore WordPressVIPMinimum.JS.HTMLExecutingFunctions.html
+												init_translations();
+											break;
+											case 'flag': // Flag in front of the select dropdown
+												// Data is built and come from server side and is well escaped when necessary
+												$( '.pll-select-flag' ).html( this.data ); // phpcs:ignore WordPressVIPMinimum.JS.HTMLExecutingFunctions.html
+											break;
+										}
+									}
+								);
+								blockEditorSavePostAndReloadPage();
 							}
 						);
-						blockEditorSavePostAndReloadPage();
-					}
+					},
+					() => {} // Do nothing when promise is rejected by clicking the Cancel dialog button.
```

Comment:
The Promise's reject callback does nothing both in block and classic editors. Do we need a reject callback at all?

Reply:
Yes! It's just to avoid a javascript error that says when uncatch a promise.